### PR TITLE
tests: bluetooth: tester: Add support for Central Address Resolution

### PR DESCRIPTION
**Pull request by sjanc**

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: tests/bluetooth/tester/src/bttester.h [hunk 1]
```diff
@@ -228,12 +228,15 @@ struct gap_passkey_confirm_cmd {
 	uint8_t match;
 } __packed;
 
+#define GAP_START_DIRECTED_ADV_HD	0x0001
+#define GAP_START_DIRECTED_ADV_OWN_ID	0x0002
+#define GAP_START_DIRECTED_ADV_PEER_RPA	0x0004
```
Comment (Thalley) [inline tests/bluetooth/tester/src/bttester.h:233]: Suggest to use BIT() for bitfields:
```suggestion
#define GAP_START_DIRECTED_ADV_HD	BIT(0)
#define GAP_START_DIRECTED_ADV_OWN_ID	BIT(1)
#define GAP_START_DIRECTED_ADV_PEER_RPA	BIT(2)
```

File: tests/bluetooth/tester/src/gap.c [hunk 2]
```diff
@@ -494,15 +541,30 @@ static void start_directed_advertising(const uint8_t *data, uint16_t len)
 	const struct gap_start_directed_adv_cmd *cmd = (void *)data;
 	struct gap_start_directed_adv_rp rp;
 	struct bt_le_adv_param adv_param;
+	uint16_t options = sys_le16_to_cpu(cmd->options);
```
Comment (Thalley) [inline tests/bluetooth/tester/src/gap.c:544]:
```suggestion
	const uint16_t options = sys_le16_to_cpu(cmd->options);
```

File: tests/bluetooth/tester/src/gap.c [hunk 3]
```diff
@@ -41,6 +42,52 @@ static struct bt_le_oob oob_sc_local = { 0 };
 static struct bt_le_oob oob_sc_remote = { 0 };
 #endif /* !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY) */
 
+#if defined(CONFIG_BT_PRIVACY)
+static struct {
+	bt_addr_le_t addr;
+	uint8_t supported;
```
Comment (Thalley) [inline tests/bluetooth/tester/src/gap.c:48]: Consider using `bool` for this

**Review by Thalley — CHANGES_REQUESTED** [review]

**Review by Thalley — CHANGES_REQUESTED** [review]

File: tests/bluetooth/tester/src/gap.c [hunk 4]
```diff
@@ -41,6 +42,52 @@ static struct bt_le_oob oob_sc_local = { 0 };
...
+	struct bt_conn_info info;
+	bool supported = false;
+
+	if (!err && data && length == 1) {
+		const uint8_t *tmp = data;
+
+		/* only 0 or 1 are valid values */
+		if (tmp[0] == 1) {
+			supported = tmp[0];
+		}
```
[the remark this answers is not in the document]
Comment (sjanc) [reply tests/bluetooth/tester/src/gap.c:75]: I'd prefer to leave explicit check but I'll change assignment to supported = true;

File: tests/bluetooth/tester/src/gap.c (hunk 2 above)
Comment (Thalley) [reply tests/bluetooth/tester/src/gap.c:544]: @sjanc Are you opposed to this change?

Comment (sjanc) [reply tests/bluetooth/tester/src/gap.c:544]: it isn't too common to constify non-pointer variables,  but I can do that if it matters

Comment (Thalley) [reply tests/bluetooth/tester/src/gap.c:544]: That's true. It's OK to leave as is, but I still see it as an improvement.

**Review by Thalley — APPROVED** [review]
LGTM

**Pull request merged**